alt_e2eshark/e2e_testing/backends.py

```diff
@@ -76,16 +76,25 @@ def func(x):
 
 class CLIREEBackend(BackendBase):
     '''This backend calls iree through the command line to compile and run MLIR modules'''
-    def __init__(self, *, device="local-task", hal_target_backend="llvm-cpu", extra_args : List[str] = None):
+    def __init__(self, *, device="local-task", hal_target_backend="llvm-cpu", target_chip = None, extra_args : List[str] = None):
         self.device = device
         self.hal_target_backend = hal_target_backend
+        self.target_chip = target_chip
         self.extra_args = []
         if extra_args:
             for a in extra_args:
                 if a[0:2] == "--":
                     self.extra_args.append(a)
                 else:
                     self.extra_args.append("--" + a)
+        elif hal_target_backend == "rocm":
```

Comment:
Maybe this should just be an "if" and instead of overwriting the extra args, it appends to it? Same w/ below

Reply:
Good point missed it in copy pasting the blocks from elsewhere